### PR TITLE
chore: Add release-snapshot script

### PR DESCRIPTION
**Pull request by izaera**
I'm copying the release-snapshot script from JS Toolkit to be able to automate snapshot releases and make the process less error prone.

### CODE REVIEW

**Comment by izaera** [comment]
I'm merging this because it's blocking.

File: scripts/release-snapshot.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+	if (!(await isStatusClean(status))) {
+		throw 'Working copy has local changes';
+	}
+
+	success('Working copy is clean');
+
+	const branch = await git('rev-parse', '--abbrev-ref', 'HEAD');
+
+	if (branch != RELEASE_BRANCH) {
```
Comment (wincent) [inline scripts/release-snapshot.js]: `!=` here but `!==` below. You should use strict equality consistently, I think, and we'll be [enforcing that via lint "soon"](https://github.com/liferay/eslint-config-liferay/issues/158).

File: scripts/release-snapshot.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+		throw `Only releases from branch ${RELEASE_BRANCH} allowed`;
+	}
+
+	success('Branch name is OK');
+
+	try {
+		await git('pull', '--ff-only', 'upstream', branch);
```
Comment (wincent) [inline scripts/release-snapshot.js]: You're assuming that people have an `upstream` remote.

Just sayin'.

File: scripts/release-snapshot.js (hunk 1 above)
Comment (izaera) [reply scripts/release-snapshot.js]: I'm removing all these in favour of liferay-js-publish, so it doesn't matter any more :man_shrugging:

**Pull request merged**